RNWCPP/Desktop/WebSocket.cpp

```diff
@@ -251,13 +251,13 @@ void BaseWebSocket<Protocol, Socket, Resolver>::Stop()
 template<typename Protocol, typename Socket, typename Resolver>
 void BaseWebSocket<Protocol, Socket, Resolver>::EnqueueWrite(const string& message, bool binary)
 {
+  post(m_context, [this, message = std::move(message), binary]()
+  {
     m_writeRequests.emplace(std::move(message), binary);
-    if (!m_writeInProgress && ReadyState::Open == m_readyState)
-      post(m_context, [this]()
-      {
-        if (!m_writeRequests.empty())
-          PerformWrite();
-      });
+
+    if (!m_writeInProgress && ReadyState::Open == m_readyState && !m_writeRequests.empty())
```

Comment:
!m_writeRequests.empty() shouldn't be needed since you are adding an element right before. Can you remove that check?

Reply:
I’ll add comments to ensure access to the write queue is always synchronized instead.